core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -2604,11 +2604,22 @@ object ZIO extends ZIOCompanionPlatformSpecific {
    * composite fiber that produces a list of their results, in order.
    */
   def forkAll[R, E, A](as: Iterable[ZIO[R, E, A]]): URIO[R, Fiber.Synthetic[E, List[A]]] =
-    as.foldRight[URIO[R, Fiber.Synthetic[E, List[A]]]](succeedNow(Fiber.succeed(Nil))) { (aIO, asFiberIO) =>
-      asFiberIO.zipWith(aIO.fork) {
-        case (asFiber, aFiber) =>
-          asFiber.zipWith(aFiber)((as, a) => a :: as)
-      }
+    as.toList match {
```

Comment:
Can we use `as.reduceRightOption(???).getOrElse(ZIO.succeedNow(Fiber.succeed(Nil)))` to clean this up?

Reply:
I've just refactored the code as you proposed.

Reply:
Looks great!